src/components/signupForm/SignUpForm.tsx

```diff
@@ -42,7 +48,7 @@ function SignupForm({ onSubmit }: SignupFormProps) {
     setConfirmPassword(e.target.value);
   };
 
-  const handleSubmit = (e: React.FormEvent) => {
+  const handleSubmit = async (e: React.FormEvent) => {
```

Comment:
handleSubmit for signup looks good. Good to type it the same way as in login + its narrower type

```typescript
const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => // ... 
```